src/clr-angular/popover/dropdown/dropdown-trigger.ts

```diff
@@ -42,8 +45,24 @@ export class ClrDropdownTrigger {
     return this.ifOpenService.open;
   }
 
+  get setAriaLabel() {
+    return this.active ? this.clrAriaLabelDropdownClose : this.clrAriaLabelDropdownOpen;
+  }
+
   @HostListener('click', ['$event'])
   onDropdownTriggerClick(event: any): void {
     this.ifOpenService.toggleWithEvent(event);
   }
+
+  @Input('clrAriaLabelDropdownOpen') clrAriaLabelDropdownOpen: string;
```

Comment:
What we can do is the same as for the Alert close button PR: https://github.com/vmware/clarity/pull/3662/files#diff-c844262cb1b2d10864f118f627115edcR31

`@Input() clrDropdownOpenAriaLabel: string = this.commonStrings.keys.dropdownOpenAriaLabel;` and you don't have to have the `ngOnInit` here. Anything locally provided will override the defaults then.

Reply:
I changed the way i set the default value because the tests were failing. If you have the case when you pass a string variable that is undefined to the input ( like in the test ), the default value wont be set. 
I see this assigning ```@Input() clrDropdownOpenAriaLabel: string = this.commonStrings.keys.dropdownOpenAriaLabel``` 
only works if you dont use the input in html at all. But if you want to set the aria label in the case i told you above, the solution i found is using onInit.

Reply:
I'm not sure that understand the issue here when it comes to the tests and testing the inputs you need two simple tests.

* test to see if the component/directive will set the default value when you don't pass anything to it. ( `[clrDropdownOpenAriaLabel]="undefined"` could be seen as not passed at all )
* test to see that when passing `clrDropdownOpenAriaLabel` it will change the `aria-label` at the dom. 

In both cases, checking to see that the `this.clrDropdownOpenAriaLabel`, is not enough. We need to be sure that the correct value is set at the DOM.

This way you gonna check the following things:

* Input exists and is working
* `ariaLabel` is setting the correct string
* host element is getting the correct `aria-label` value.

Reply:
Thanks for the tips! The problem is that `[clrDropdownOpenAriaLabel]="undefined"`  isnt the same thing as not passing a value. If you pass any reference to the input, that will overrides the default set. So the directive will never set a default value if there s a null or undefined reference. One solution to catch that case is to use ngOnInit. 
What i wanted to say is that the tests for the default value will always crash in this case.

Reply:
Tell me if i get it correctly. Should I consider that there wont be inputs that are undefined?  And also make tests for inputs to be expected as defined?